js/src/widgets/metadataView.js

```diff
@@ -208,30 +219,77 @@
   },
 
     bindEvents: function() {
+      var _this = this;
+      jQuery.subscribe('ANNOTATIONS_LIST_UPDATED', function(event, options) {
+        _this.updateMetadata(options);
+      });
     },
 
     toggle: function(stateValue) {
-        if (stateValue) { 
-            this.show(); 
+        if (stateValue) {
+            this.show();
         } else {
             this.hide();
         }
     },
-    
+
```

Comment:
This is where most of the logic is: find metadata annotations for the canvasid we have now and use it to populate the metadata.